desktop/core/src/desktop/js/apps/storageBrowser/StorageBrowserPage/StorageBrowserTable/StorageBrowserTable.tsx

```diff
@@ -0,0 +1,150 @@
+// Licensed to Cloudera, Inc. under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  Cloudera, Inc. licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { useEffect, useState } from 'react';
+import { i18nReact } from '../../../../utils/i18nReact';
+import Table from 'cuix/dist/components/Table';
+import { ColumnProps } from 'antd/lib/table';
+import FolderIcon from '@cloudera/cuix-core/icons/react/ProjectIcon';
+//TODO: Use cuix icon (Currently fileIcon does not exist in cuix)
+import { FileOutlined } from '@ant-design/icons';
+
+import { PageStats, StorageBrowserTableData } from '../../../../reactComponents/FileChooser/types';
+import StorageBrowserPagination from './StorageBrowserPagination/StorageBrowserPagination';
+import './StorageBrowserTable.scss';
+import Tooltip from 'antd/es/tooltip';
+
+interface StorageBrowserTableProps {
+  className?: string;
+  dataSource: StorageBrowserTableData[];
+  onFilepathChange: (path: string) => void;
+  onPageNumberChange: (pageNumber: number) => void;
+  onPageSizeChange: (pageSize: number) => void;
+  pageSize: number;
+  pageStats: PageStats;
+  rowClassName?: string;
+  testId?: string;
+}
+
+const defaultProps = {
+  className: 'hue-storage-browser__table',
+  rowClassName: 'hue-storage-browser__table-row',
+  testId: 'hue-storage-browser__table'
+};
+
+const StorageBrowserTable: React.FC<StorageBrowserTableProps> = ({
+  className,
+  dataSource,
+  onFilepathChange,
+  onPageNumberChange,
+  onPageSizeChange,
+  pageSize,
+  pageStats,
+  rowClassName,
+  testId,
+  ...restProps
+}): JSX.Element => {
+  const [tableHeight, setTableHeight] = useState<number>();
+
+  const { t } = i18nReact.useTranslation();
+
+  const getColumns = file => {
+    const columns: ColumnProps<unknown>[] = [];
+    for (const [key] of Object.entries(file)) {
+      const temp: ColumnProps<unknown> = {
+        dataIndex: `${key}`,
+        title: t(`${key}`.charAt(0).toUpperCase() + `${key}`.slice(1)),
+        key: `${key}`
+      };
+      if (key === 'name') {
+        // temp.ellipsis = true;
+        temp.width = '45rem';
```

Comment:
How does this work, will the name column always be 45rem in width? Why can't it go below if the available space is lacking, and what if we have more space available and we still cut of the name after 45rem.. seems unnecessary to me, or is that according to the design specs?

Reply:
This is actually according to the current hue specifics we have where each column is assigned a percentage of space it has to occupy(name - 45%, last updated -15%, rest -10%). I referenced the same and implemented that.

Reply:
But "45rem" is not the same as 45% of the available width...